ovs/src/main/java/org/dcsa/conformance/standards/ovs/checks/OvsChecks.java

```diff
@@ -0,0 +1,300 @@
+    checks.add(JsonAttribute.customValidator(
+      "Validate limit parameter is met",
+      body -> {
+        Optional<Map.Entry<OvsFilterParameter, String>> limitParam =
+          sspSupplier.get().getMap().entrySet().stream()
+            .filter(e -> e.getKey().equals(OvsFilterParameter.LIMIT))
+            .findFirst();
+
+        if (limitParam.isPresent()) {
+          int expectedLimit = Integer.parseInt(limitParam.get().getValue().trim());
+          if (body.size() > expectedLimit) {
+            return Set.of("The number of events exceeds the limit parameter: " + expectedLimit);
+          }
+        }
+
+        return Set.of();
+      }
+    ));
+
+    return JsonAttribute.contentChecks(
+      OvsRole::isPublisher,
+      matched,
+      HttpMessageType.RESPONSE,
+      standardVersion,
+      checks
+    );
+  }
+
+  private Set<String> validateParameter(JsonNode body, Supplier<SuppliedScenarioParameters> sspSupplier,
+                                        OvsFilterParameter parameter, String... jsonPaths) {
+    Optional<Map.Entry<OvsFilterParameter, String>> param = sspSupplier.get().getMap().entrySet().stream()
+      .filter(e -> e.getKey().equals(parameter))
+      .findFirst();
+
+    if (param.isPresent()) {
+      Set<String> expectedValues = Arrays.stream(param.get().getValue().split(","))
+        .map(String::trim)
+        .collect(Collectors.toSet());
+
+      // Check if ANY of the jsonPaths match the expectedValues
+      if(jsonPaths.length > 1) {
+        boolean anyMatch = Stream.of(jsonPaths)
+          .anyMatch(jsonPath ->
+            findMatchingNodes(body, jsonPath)
+              .anyMatch(valueNode -> !(valueNode.isMissingNode() || valueNode.isNull())
+                && expectedValues.contains(valueNode.asText())));
+
+        if (!anyMatch) {
+          return Set.of("Missing or mismatched values for parameter: " + parameter.getQueryParamName()
+            + " in any of the paths: " + Arrays.toString(jsonPaths));
+        }
+      }
+      else {
+        Set<String> errors = Stream.of(jsonPaths)
+          .flatMap(jsonPath ->
+            findMatchingNodes(body, jsonPath)
+              .filter(valueNode -> !(valueNode.isMissingNode() || valueNode.isNull())
+                && !expectedValues.contains(valueNode.asText()))
+              .map(valueNode -> "Missing or mismatched " + jsonPath + ": " + valueNode.asText()))
+          .collect(Collectors.toSet());
+
+        return errors.isEmpty() ? Set.of() : errors;
+      }
+    }
+    return Set.of();
+  }
+
+  private Set<String> validateDate(
+    JsonNode body,
+    Supplier<SuppliedScenarioParameters> sspSupplier,
+    OvsFilterParameter dateParameter,
+    BiPredicate<LocalDate, LocalDate> dateComparison) {
+
+    Optional<Map.Entry<OvsFilterParameter, String>> dateParam =
+      sspSupplier.get().getMap().entrySet().stream()
+        .filter(e -> e.getKey().equals(dateParameter))
+        .findFirst();
+
+    if (dateParam.isPresent()) {
```

Comment:
Perhaps you can return here when the dateParam is not present. That avoids many indented lines and mental effort. 
See for background info: https://medium.com/swlh/return-early-pattern-3d18a41bba8

Reply:
The logic is if the dateParams is provided in the query parameters then the check must be performed otherwise nothing must be performed. why should i waste the runtime cycle to check if dateParam is not present for every scenarios and return empty set.   I find this logic must only be executed when the dateParam is present.